compiler/rustc_mir_build/src/thir/pattern/const_to_pat.rs

```diff
@@ -445,20 +437,17 @@ impl<'tcx> ConstToPat<'tcx> {
                         // we fall back to a const pattern. If we do not do this, we may end up with
                         // a !structural-match constant that is not of reference type, which makes it
                         // very hard to invoke `PartialEq::eq` on it as a fallback.
-                        let val = match self.recur(tcx.deref_mir_constant(self.param_env.and(cv)), false) {
-                            Ok(subpattern) => PatKind::Deref { subpattern },
-                            Err(_) => PatKind::Constant { value: cv },
-                        };
+                        let subpattern = self.recur(tcx.deref_mir_constant(self.param_env.and(cv)), false)?;
                         self.behind_reference.set(old);
-                        val
+                        PatKind::Deref { subpattern }
```

Comment:
This is just a more general case of the example already given, as it also works if you put something like a tuple or other builtin aggregate in between.

Reply:
This is the fishiest reasoning of all of them, but everything I've checked out already has had tests or was just a variant using a different aggregate